src/Microsoft.AspNet.Mvc.ViewFeatures/Resources.resx

```diff
@@ -259,4 +259,7 @@
   <data name="RemoteAttribute_RemoteValidationFailed" xml:space="preserve">
     <value>'{0}' is invalid.</value>
   </data>
+  <data name="HtmlGenerator_FieldNameCannotBeNullOrEmpty" xml:space="preserve">
+    <value>The field name for Html field cannot be Null or Empty. Please provide a non-Empty field name OR use 'EditorFor' as a workaround with parameter name=null.</value>
```

Comment:
Suggest `The name of an HTML field cannot be null or empty. Instead use methods {1}.{2} or {3}.{4} with a non-empty {5} argument value.` and pass `typeof(IHtmlHelper).FullName`, `nameof(IHtmlHelper.Editor)`, `typeof(IHtmlHelper<>).FullName`, `nameof(IHtmlHelper<>.EditorFor)`, `"htmlFieldName"`.